$OpenBSD: patch-bootstrap,v 1.10 2015/12/08 17:18:59 dcoppa Exp $

Use Ninja to build CMake
XXX: broken on arm and m88k

--- bootstrap.orig	Thu Nov 12 16:39:53 2015
+++ bootstrap	Mon Nov 23 16:54:15 2015
@@ -158,7 +158,11 @@ if ${cmake_system_mingw}; then
   cmake_bootstrap_generator="MSYS Makefiles"
 else
   # Bootstrapping from a standard UNIX prompt.
-  cmake_bootstrap_generator="Unix Makefiles"
+  if arch -s | egrep -q '^arm|^m88k'; then
+    cmake_bootstrap_generator="Unix Makefiles"
+  else
+    cmake_bootstrap_generator="Ninja"
+  fi
 fi
 
 # Choose tools and extensions for this platform.
@@ -322,6 +326,11 @@ CMAKE_CXX_SOURCES="\
   cmExprLexer \
   cmExprParser \
   cmExprParserHelper \
+  cmGlobalNinjaGenerator \
+  cmLocalNinjaGenerator \
+  cmNinjaTargetGenerator \
+  cmNinjaNormalTargetGenerator \
+  cmNinjaUtilityTargetGenerator \
 "
 
 if ${cmake_system_mingw}; then
